webui/react/src/stores/workspaces.tsx

```diff
@@ -53,15 +60,40 @@ export const useEnsureWorkspacesFetched = (canceler: AbortController): (() => Pr
     } catch (e) {
       handleError(e);
     }
-  }, [canceler, workspaces, updateWorkspaces]);
+  }, [canceler, workspaces]);
 };
 
-export const useWorkspaces = (): Loadable<Workspace[]> => {
+export const useWorkspaces = (params?: GetWorkspacesParams): Loadable<Workspace[]> => {
   const context = useContext(WorkspacesContext);
   if (context === null) {
     throw new Error('Attempted to use useWorkspaces outside of Workspace Context');
   }
-  const { workspaces } = context;
+  if (context.workspaces === NotLoaded) {
```

Comment:
Nit but I would just use `Loadable.match` here.

Reply:
Or just `Loadable.map` since the `NotLoaded` case is just `NotLoaded`.

Reply:
✅